UnityProject/Assets/Scripts/Systems/Electricity/Electrical processes/ElectricalManager.cs

```diff
@@ -59,7 +59,15 @@ private void UpdateMe()
 				{
 					lock (ElectricalLock)
 					{
-						electricalSync.PowerNetworkUpdate();
+						try
+						{
+							electricalSync.PowerNetworkUpdate();
+						}
+						catch (Exception e)
+						{
+							Logger.LogError($"Electrical MainThreadProcess Error! \n{e}", Category.Electrical);
```

Comment:
include the trace

Reply:
apparently you can just put `e` or `e.ToString()` and it'll put the whole message with stacktrace

Reply:
looked into the unity code for it and gagged, so ive made an extension instead